### ISSUE TITLE
Accessibility

### ISSUE THREAD

**Issue by doana** [artefactual/atom#1509]
Hi folks, 

I wanted to bring the issue of web accessibility to your attention for this project as there are a few low-hanging fruit that could be addressed in order to create a much better experience for users of assistive technology like screen readers. 

Our local legislation references an international accessibility standard - The [Web Content Accessibility Guidelines (WCAG) version 2.1](https://www.w3.org/TR/WCAG21/). This document would be a great resource for the implementation of accessibility features in AtoM.  

In particular, the most prevalent issues that I saw in my quick checking were: 
1. Low [contrast](https://www.w3.org/WAI/WCAG21/Understanding/contrast-minimum.html) between text and its background in the Dominion theme. I haven't checked the Archives Canada theme for this issue. Examples of this issue include the link color, orange h2's, light grey h3's.
2. A lack of [labels for form fields](https://www.w3.org/WAI/WCAG21/Techniques/general/G131.html) sporadically throughout the application including the search and advanced search forms

If these issues could be addressed, it would go a long way toward creating a better experience for all users of AtoM.

**Comment by fiver-watson** [comment]
Hi @doana , 

Thanks for raising this issue. It's one we are aware of, and we do have some very relevant work coming soon, described further below. First, I just wanted to mention that thus far, the greatest challenge for us in doing work like this to enhance AtoM has been our current business model. We have always strived for openness, and release not only AtoM under an open / libre license, but all supporting materials as well like our documentation, presentations and slides, wiki content, forum support, and more. Consequently, while our team invests a lot of time and effort in addressing bugs, maintenance upgrades, and performance improvements with each release, we have relied on our community to support feature enhancements and major dependency upgrades, either in the form of code contributions or via community development sponsorship. You can read about the history and development philosophy of the AtoM project thus far on our wiki, here: 

* https://wiki.accesstomemory.org/Development/Philosophy

Artefactual is in the process of trying to evolve our business model so we can more easily maintain and develop our applications (and create new ones), without compromising on our values and commitment to openness. This is a slow process that will take time to implement - so in the meantime, if you'd like to submit a pull request addressing some of these elements, we have some guidance [on our wiki](https://wiki.accesstomemory.org/Development#Development_resources); or if your institution might be interested in sponsoring accessibility development in AtoM, feel free to contact our team at info@artefactual.com to discuss options. 

That said: 

We have been working on a major upgrade of AtoM's CSS library, from Bootstrap version 2.2 to [version 5.1](https://getbootstrap.com/). Later versions of Bootstrap have focused heavily on improving both responsiveness and accessibility support, from which AtoM will benefit greatly. We don't have in-house accessibility testing experts at Artefactual, but our hope is that these changes should bring AtoM either up to or at least much closer to WCAG 2.1 AA-level compliance. 

Part of the reason that our 2.7 release has been delayed has been trying to work out the best way to make this development work available to our community and our hosted clients without major disruption - since a total overhaul of AtoM's user interface will almost certainly break any existing custom themes. There are also a lot of other enhancements and features slated for 2.7, including a move to PHP 7.4 and Ubuntu 20.04, so we have been trying to determine whether to prioritize a 2.7 release as soon as possible and save the BS5 upgrade for the next major release, or include everything in the next major release. Either way, when this work is included it will be a transitional release, including both BS5 and BS2 templates, so local users have time to address custom themes, etc. In light of this, we also need to develop a deprecation plan for finally removing any remaining BS2 components. 

In the meantime, if you'd like to access this work immediately (despite the fact that it is not yet considered a stable release), it's all available in the following development branch: 

* https://github.com/artefactual/atom/tree/dev-test/issue-13530-b5-themes

You can create a local development environment by following [Option 2](https://www.accesstomemory.org/docs/2.7/admin-manual/installation/ubuntu/) (install from our code repository) in the 2.7 Installation docs, using this branch. 

One of the most helpful ways to support this work would be to provide accessibility testing and evaluation, if you have anyone locally with the tools and expertise to do so. If your institution does have an accessibility tester but you can't set up your own test environment, feel free to contact me off-thread ( dan [at] artefactual [dot] com ) and we can discuss options for Artefactual to potentially provide you with access to a test instance. 

In the meantime, vanilla 2.7 (using the current Bootstrap version 2.2) has also received some minor accessibility enhancements via a community contributor. These are all in the main qa/2.x development branch where we prepare upcoming releases. A summary of the related issues can be seen on the roadmap page in our wiki, here: 

* https://wiki.accesstomemory.org/Releases/Roadmap#Accessibility
* Related issue ticket: https://projects.artefactual.com/issues/13535

For contrast and other layout / UI issues, these could be addressed locally via a custom theme. We have some resources on how to develop these, which I recently shared in this AtoM User Forum thread: 

* https://groups.google.com/g/ica-atom-users/c/BnqcZJcwgE0/m/LFQ_QPv0AQAJ

Hope this helps! 

Dan Gillean, MAS MLIS
AtoM Program Manager
Artefactual Systems

**Comment by doana** [comment]
Dan, thanks so much for such a thorough response. I'll have to dig through these resources and see if I can find some time to do some more detailed accessibility testing on the 2.7 branch. 

Do you have a rough release date for 2.7?

**Comment by fiver-watson** [comment]
We're still working out a release plan. It will be this year - and hopefully this quarter, but I'd say that next quarter is more realistic.

**Comment by dandalpiaz** [comment]
It looks like there have been a couple new releases since this issue was discussed - wanted to add that a few things are still being flagged in the default theme, arDominionB5Plugin. For example, on the demo site, https://demo.accesstomemory.org/informationobject/browse?topLod=0&sort=relevance&repos=&query= 

- Background and foreground colors do not have a sufficient contrast ratio - this is flagged on the orange text, e.g. "CA ON00012 SC034"
- Links rely on color to be distinguishable - this is flagged on the "Read more" links, likely due to their color / smaller size, and lack of underline / differentiating style.

These were found using the 'Lighthouse' accessibility checker built in to the Chrome browser, which utilizes the Deque's axe-core engine, which can also be utilized in many other tools, https://github.com/dequelabs/axe-core/blob/develop/doc/projects.md . 

Addressing accessibility issues will become a more pressing matter since a recent update to Title II of the Americans with Disabilities Act (ADA) has specified that WCAG 2.1, Level AA is the standard that needs to be followed for state and local governments in the US (which will include public universities, libraries, etc.) More info here, https://www.ada.gov/notices/2024/03/08/web-rule/

**Comment by sarah-mason** [comment]
Hi @doana - again, thank you for your issue! I notice this is really a two-part request and part 1 has been addressed in PR #1840 almost a year ago. We try to keep issues singular as much as possible to keep track of the issue and associated PRs. If part two is an issue, I encourage that you log a new issue with details so we (or another contributor) can address it and link the PR to it. I will close this issue for now to reflect some of this issue is addressed.
